docs/devel/owners.md

```diff
@@ -0,0 +1,128 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/devel/owners.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Owners files
+
+_Note_: This is a design for a feature that is not yet implemented.
+
+## Overview
+
+We want to establish owners for different parts of the code in the Kubernetes codebase.  These owners
+will serve as the approvers for code to be submitted to these parts of the repository.  Notably, owners
+are not necessarily expected to do the first code review for all commits to these areas, but they are
+required to approve changes before they can be merged.
+
+## High Level flow
+
+### Step One: A PR is submitted
+
+After a PR is submitted, the automated kubernetes PR robot will append a message to the PR indicating the owners
+that are required for the PR to be submitted.
+
+Subsequently, a user can also request the approval message from the robot by writing:
+
+```
+@k8s-bot approvers
+```
+
+into a comment.
+
+In either case, the automation replies with an annotation that indicates
+the owners required to approve.  The annotation is a comment that is applied to the PR.
+This comment will say:
+
+```
+Approval is required from <owner-a> OR <owner-b>, AND <owner-c> OR <owner-d>, AND ...
+```
+
+The set of required authors is drawn from the OWNERS files in the repository (see below).  For each file
+there should be multiple different OWNERS, these owners are listed in the `OR` clause(s). Because
+it is possible that a PR may cover different directories, with disjoint sets of OWNERS, a PR may require
+approval from more than one person, this is where the `AND` clauses come from.
+
+### Step Two: A PR is LGTM'd
+
+Once a PR is reviewed and LGTM'd it is eligible for submission.  However, for it to be submitted
+an owner for all of the files changed in the PR have to 'approve' the PR.  A user is an owner for a
+file if they are included in the OWNERS hierarchy (see below) for that file.
+
+Owner approval comes in two forms:
+
+   * An owner adds a comment to the PR saying "I approve"
```

Comment:
I meant "LGTM" as comment rather than "I approve"

On Tue, Nov 17, 2015 at 9:38 PM, Eric Paris <notifications@github.com>
wrote:

> In docs/devel/owners.md
> <https://github.com/kubernetes/kubernetes/pull/17267#discussion_r45161784>
> :
>
> > +```
> > +
> > +The set of required authors is drawn from the OWNERS files in the repository (see below).  For each file
> > +there should be multiple different OWNERS, these owners are listed in the `OR` clause(s). Because
> > +it is possible that a PR may cover different directories, with disjoint sets of OWNERS, a PR may require
> > +approval from more than one person, this is where the `AND` clauses come from.
> > +
> > +### Step Two: A PR is LGTM'd
> > +
> > +Once a PR is reviewed and LGTM'd it is eligible for submission.  However, for it to be submitted
> > +an owner for all of the files changed in the PR have to 'approve' the PR.  A user is an owner for a
> > +file if they are included in the OWNERS hierarchy (see below) for that file.
> > +
> > +Owner approval comes in two forms:
> > +
> > +   * An owner adds a comment to the PR saying "I approve"
>
> My first thought was a LGTM-owner label, but changes which spam areas
> will need multiple owner approvals and the only way I know of to do that is
> in comments.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17267/files#r45161784>.
>